pkg/controllers/mcs/service_export_controller.go

```diff
@@ -466,7 +483,48 @@ func (c *ServiceExportController) removeOrphanWork(ctx context.Context, endpoint
 			continue
 		}
 
-		err := cleanEndpointSliceWork(ctx, c.Client, &work)
+		orphanEndpointSliceWorks = append(orphanEndpointSliceWorks, &work)
+	}
+
+	if len(orphanEndpointSliceWorks) == 0 {
+		return nil
+	}
+
+	dynamicClusterClient, err := c.ClusterDynamicClientSetFunc(serviceExportKey.Cluster, c.Client, c.ClusterClientOption)
+	if err != nil {
+		klog.ErrorS(err, "Failed to build dynamic cluster client", "cluster", serviceExportKey.Cluster)
+		return err
+	}
+
+	// Before deleting the EndpointSlice Work, perform a secondary confirmation by directly accessing the apiserver
+	// of member cluster to verify whether the target EndpointSlice resource exists.
+	var errs []error
+	for index := range orphanEndpointSliceWorks {
+		orphanEpsWork := orphanEndpointSliceWorks[index]
+
+		// To prevent a panic caused by the absence of the index during the unmarshalling of the Work Manifests.
```

Comment:
```suggestion
		// This check avoids a potential panic due to a nil reference when accessing the Work Manifests.
		// It can also avoid annoying false positive lint checks.
```